src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/DSA.cs

```diff
@@ -79,27 +79,66 @@ public byte[] SignData(byte[] data, HashAlgorithmName hashAlgorithm)
             {
                 throw new ArgumentNullException(nameof(data));
             }
+
             return SignData(data, 0, data.Length, hashAlgorithm);
         }
 
+        public byte[] SignData(byte[] data, HashAlgorithmName hashAlgorithm, DSASignatureFormat signatureFormat)
+        {
+            if (data == null)
+            {
+                throw new ArgumentNullException(nameof(data));
+            }
+
+            return SignDataCore(data, hashAlgorithm, signatureFormat);
+        }
+
         public virtual byte[] SignData(byte[] data, int offset, int count, HashAlgorithmName hashAlgorithm)
         {
             if (data == null) { throw new ArgumentNullException(nameof(data)); }
             if (offset < 0 || offset > data.Length) { throw new ArgumentOutOfRangeException(nameof(offset)); }
             if (count < 0 || count > data.Length - offset) { throw new ArgumentOutOfRangeException(nameof(count)); }
             if (string.IsNullOrEmpty(hashAlgorithm.Name)) { throw HashAlgorithmNameNullOrEmpty(); }
 
-            byte[] hash = HashData(data, offset, count, hashAlgorithm);
-            return CreateSignature(hash);
+            return SignDataCore(data, hashAlgorithm, DSASignatureFormat.IeeeP1363FixedFieldConcatenation);
+        }
+
+        public byte[] SignData(byte[] data, int offset, int count, HashAlgorithmName hashAlgorithm, DSASignatureFormat signatureFormat)
+        {
+            if (data == null) { throw new ArgumentNullException(nameof(data)); }
+            if (offset < 0 || offset > data.Length) { throw new ArgumentOutOfRangeException(nameof(offset)); }
+            if (count < 0 || count > data.Length - offset) { throw new ArgumentOutOfRangeException(nameof(count)); }
+
+            return SignDataCore(new ReadOnlySpan<byte>(data, offset, count), hashAlgorithm, signatureFormat);
+        }
+
+        protected virtual byte[] SignDataCore(ReadOnlySpan<byte> data, HashAlgorithmName hashAlgorithm, DSASignatureFormat signatureFormat)
```

Comment:
In a manner left as an exercise to the reader: I'm concerned that the current changes won't work for existing out-of-box implementations that override the current virtuals.  Make sure there's a test that calling the new method specifying the old format does sensible things with netstandard2.1-based derived types.

Reply:
I believe they should work since I always end up calling abstract method as a fallback if it's not overriden (there is even an existing test which tests some of that scenarios)

Reply:
> there is even an existing test which tests some of that scenarios

Oh, good.  Yeah, I hadn't made it through the call chain yet to see that there was that fallback and if it was going to be complicated wanted to get the feedback out there before I had to run.